### ISSUE TITLE
Many Iterator allocations because of foreach loops.

### ISSUE THREAD

**Issue by denisk20** [Trixt0r/spriter#20]
Profiling on Android shows that there are many Iterator allocations which happen because of `foreach` loops executed on every frame. Details here, in **Avoid Iterators** section:
https://medium.com/google-developers/developing-for-android-ii-bb9a51f8c8b9

This is a chart for running one simple animation for just a few seconds:
![image](https://cloud.githubusercontent.com/assets/3080318/10586203/bc32da94-76a3-11e5-8bfa-e5ab52724e21.png)

![image](https://cloud.githubusercontent.com/assets/3080318/10585963/a4f77d0e-76a2-11e5-8189-9f98ad1d7361.png)

Basically almost all outer circle of the chart corresponds to `Iterator` allocations.

Replacing `foreach` loops with indexed `for` loops in `Player.update()` and some other places fixes this issue:

![image](https://cloud.githubusercontent.com/assets/3080318/10586586/869c9648-76a5-11e5-8619-86fa8f9ddecf.png)

There are no iterator allocations here at all.
Let me know if you're interested in a pull request for this.

**Comment by Trixt0r** [comment]
Thanks for running a profiler on Android!

I would really appreciate a pull request.

**Comment by denisk20** [comment]
@Trixt0r Pull request created, please review.
